src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/OneVersusAllTrainer.cs

```diff
@@ -586,17 +579,17 @@ private sealed class ImplDist : ImplBase
             public override IValueMapper[] Predictors => _mappers;
             public override bool CanSavePfa { get; }
 
-            internal ImplDist(IValueMapperDist[] predictors)
+            internal ImplDist(T[] predictors)
             {
                 Contracts.Check(Utils.Size(predictors) > 0);
 
                 _mappers = new IValueMapperDist[predictors.Length];
                 VectorDataViewType inputType = null;
                 for (int i = 0; i < predictors.Length; i++)
                 {
-                    var vm = predictors[i];
+                    var vm = predictors[i] as IValueMapperDist;
                     Contracts.Check(IsValid(vm, ref inputType), "Predictor doesn't implement the expected interface");
-                    _mappers[i] = vm;
+                    _mappers[i] = vm as IValueMapperDist;
```

Comment:
When you see a cast which must be successful, please use `(T) vm` rather than `vm as T`.

Reply:
Line 591 uses the `Contracts.Check` to determine if the cast was successful. In fact, the cast is already happening on line 590, so I don't need to do it here again. I will remove the cast there. Does that work for you?

Reply:
Sounds good.